client/cli/proc-macro/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+/// ```
+///
+/// # Example using the actual node name
+///
+/// ```ignore
+/// Builds a new service for a light client.
+/// #[sc_cli::substrate_cli_node_name(config.network.node_name.as_str())]
+/// pub fn new_light(config: Configuration) -> Result<TaskManager, ServiceError> {
+///     let (client, backend, keystore, mut task_manager, on_demand) =
+///         sc_service::new_light_parts::<Block, RuntimeApi, Executor>(&config)?;
+///
+///        ...
+/// }
+/// ```
+///
+/// Will produce logs that look like this:
+///
+/// ```text
+/// 2020-10-16 08:12:57  Substrate Node
+/// 2020-10-16 08:12:57  ✌️  version 2.0.0-efb9b822a-x86_64-linux-gnu
+/// 2020-10-16 08:12:57  ❤️  by Anonymous, 2017-2020
+/// 2020-10-16 08:12:57  📋 Chain specification: Local Testnet
+/// 2020-10-16 08:12:57  🏷 Node name: open-harbor-1619
+/// 2020-10-16 08:12:57  👤 Role: LIGHT
+/// 2020-10-16 08:12:57  💾 Database: RocksDb at /tmp/substrate9T9Mtb/chains/local_testnet/db
+/// 2020-10-16 08:12:57  ⛓  Native runtime: node-template-1 (node-template-1.tx1.au1)
+/// 2020-10-16 08:12:58  [open-harbor-1619] 🔨 Initializing Genesis block/state (state: 0x121d…8e36, header-hash: 0x24ef…8ff6)
+/// 2020-10-16 08:12:58  [open-harbor-1619] Loading GRANDPA authorities from genesis on what appears to be first startup.
+/// 2020-10-16 08:12:58  [open-harbor-1619] ⏱  Loaded block-time = 6000 milliseconds from genesis on first-launch
+/// 2020-10-16 08:12:58  [open-harbor-1619] Using default protocol ID "sup" because none is configured in the chain specs
+/// 2020-10-16 08:12:58  [open-harbor-1619] 🏷 Local node identity is: 12D3KooWRzmYC8QTK1Pm8Cfvid3skTS4Hn54jc4AUtje8Rqbfgtp
+/// 2020-10-16 08:12:58  [open-harbor-1619] 📦 Highest known block at #0
+/// 2020-10-16 08:12:58  [open-harbor-1619] 〽️ Prometheus server started at 127.0.0.1:9615
+/// 2020-10-16 08:12:58  [open-harbor-1619] Listening for new connections on 127.0.0.1:9944.
+/// ```
+#[proc_macro_attribute]
+pub fn substrate_cli_node_name(arg: TokenStream, item: TokenStream) -> TokenStream {
+	let item_fn = syn::parse_macro_input!(item as ItemFn);
+
+	if arg.is_empty() {
+		return Error::new(Span::call_site(), "missing expression (name of the node)")
+			.to_compile_error()
+			.into();
+	}
+
+	let name = syn::parse_macro_input!(arg as Expr);
+
+	let crate_name = if std::env::var("CARGO_PKG_NAME")
+		.expect("cargo env var always there when compiling; qed")
+		== "sc-cli"
+	{
+		Ident::new("sc_cli", Span::call_site().into())
+	} else {
+		let crate_name = match crate_name("sc-cli") {
+			Ok(x) => x,
+			Err(err) => return Error::new(Span::call_site(), err).to_compile_error().into(),
+		};
+
+		Ident::new(&crate_name, Span::call_site().into())
+	};
+
+	let ItemFn {
+		attrs,
+		vis,
+		sig,
+		block,
+	} = item_fn;
+
+	(quote! {
+		#(#attrs)*
+		#vis #sig {
+			let span = #crate_name::tracing::info_span!("substrate-node", name = #name);
+			let _enter = span.enter();
+
+			#block
+		}
+	})
+	.into()
```

Comment:
```suggestion
	quote! {
		#(#attrs)*
		#vis #sig {
			let span = #crate_name::tracing::info_span!("substrate-node", name = #name);
			let _enter = span.enter();

			#block
		}
	}.into()
```

Why did you put parentheses around this?

Reply:
It doesn't compile without

Reply:
What is the error?